### PR TITLE
Conman 703/adding field

### PR DESCRIPTION
**Pull request by JusteenRandev**
## Description

- This adds an optional field to the send email for the `engage-messaging-sendgrid` destination
- New field is ip_pool_name
- Allows sender to choose which ip pool they are sending their email from (Specifies which ip address they are using)
- If no ip pool name is given the default behaviour continues
- No pool name or a valid pool name MUST be given. If an invalid pool name is entered the email will not send

## Testing

_Include any additional information about the testing you have completed to
ensure your changes behave as expected. For a speedy review, please check
any of the tasks you completed below during your testing._

Additional manual testing was done to check the email gets sent from the correct IP address when the pool name is given.
See below screenshots from the sendgrid account (Repeated for each ip pool):
image
IPPoolDemo
Untitled
IPPoolNameNone

- [x] Added [unit tests](https://github.com/segmentio/action-destinations/blob/main/docs/testing.md#local-end-to-end-testing) for new functionality
- [x] Tested end-to-end using the [local server](https://github.com/segmentio/action-destinations/blob/main/docs/testing.md#local-end-to-end-testing)
- [ ] [Segmenters] Tested in the staging environment

### CODE REVIEW

File: packages/destination-actions/src/destinations/engage/sendgrid/__tests__/send-email.test.ts [hunk 1]
```diff
@@ -1495,6 +1496,57 @@ describe.each([
...
+        }),
+        settings,
+        mapping: getDefaultMapping({ ipPool: 'testPoolName' })
+      })
+
+      expect(responses.length).toBeGreaterThan(0)
+      expect(sendGridRequest.isDone()).toEqual(true)
+    })
+
+    it('sends the email to ip pool when name is not specified', async () => {
```
Comment (vikramkumar19) [inline packages/destination-actions/src/destinations/engage/sendgrid/__tests__/send-email.test.ts:1530]: Can we add a test case when ipPool === `''`. To make sure we capture all scenario's of ipPools `undefined` or `''`

**Review by vikramkumar19 — COMMENTED** [review]
Overall LGTM!Can we add few tests around all the possible values of ipPool. Also can you check if `null` is referred as `undefined`

File: packages/destination-actions/src/destinations/engage/sendgrid/sendEmail/SendEmailPerformer.ts [hunk 2]
```diff
@@ -193,6 +193,16 @@ export class SendEmailPerformer extends MessageSendPerformer<Settings, Payload>
       mailContent = mailContentSubscriptionHonored
       this.statsClient?.incr('request.dont_pass_subscription', 1)
     }
+    // Check if ip pool name is provided and sends the email with the ip pool name if it is
+    if (this.payload.ipPool) {
+      mailContent = {
+        ...mailContent,
+        ip_pool_name: this.payload.ipPool
```
Comment (sumitbhanwala) [inline packages/destination-actions/src/destinations/engage/sendgrid/sendEmail/SendEmailPerformer.ts:200]: @JusteenRandev I am pretty sure that you have used the right field name as documentation suggests  `ipPoolName` I also remember running into the discrepancy into the documentation vs what works in send grid API

Comment (sumitbhanwala) [reply packages/destination-actions/src/destinations/engage/sendgrid/sendEmail/SendEmailPerformer.ts:200]: Also what Vikram said if this.payload.ipPool resolves to an empty string ''. Will it evaluate to true.
If yes than we shouldn't be going inside the if block

Comment (JusteenRandev) [reply packages/destination-actions/src/destinations/engage/sendgrid/sendEmail/SendEmailPerformer.ts:200]: It works when I use that field when manually testing and I see it in the documentation here: https://docs.sendgrid.com/api-reference/mail-send/mail-send

Comment (sumitbhanwala) [reply packages/destination-actions/src/destinations/engage/sendgrid/sendEmail/SendEmailPerformer.ts:200]: Yes that's what I meant, If we use there client they do it camelCase
ipPoolName: 'transactional email'.
Probably the client will convert before sending to server.

**Review by vikramkumar19 — APPROVED** [review]
LGTM!

File: packages/destination-actions/src/destinations/engage/sendgrid/sendEmail/SendEmailPerformer.ts (hunk 2 above)
Comment (JusteenRandev) [reply packages/destination-actions/src/destinations/engage/sendgrid/sendEmail/SendEmailPerformer.ts:200]: If it resolves to an empty string it evaluates to false. I tested this manually and added it as a test case. The ip_pool_name field doesn't get set. The mailContent seems to be getting sent as a json object in a post request [here](https://github.com/segmentio/action-destinations/pull/1677/files#diff-0b5e29bcd732565f68cff10ca6765608743a263d3d4131894734d79f044f9272L201) so I followed the documentation for the curl command for the field. It's seems to work when testing manually so I'm not sure what else to check that it is the right field name.

File: packages/destination-actions/src/destinations/engage/sendgrid/sendEmail/actionDefinition.ts [hunk 3]
```diff
@@ -119,6 +119,12 @@ export const actionDefinition: ActionDefinition<Settings, Payload> = {
       type: 'boolean',
       default: false
     },
+    ipPool: {
+      label: 'IP Pool',
+      description: 'Send email with an ip pool',
+      type: 'string',
+      default: ''
```
Comment (NolanChan) [inline packages/destination-actions/src/destinations/engage/sendgrid/sendEmail/actionDefinition.ts:126]: Just wondering if there's an explicit reason to assign an empty string default? It shouldn't affect anything either way, so no need to change it

**Review by NolanChan — APPROVED** [review]

**Review by sumitbhanwala — APPROVED** [review]
LGTM

File: packages/destination-actions/src/destinations/engage/sendgrid/sendEmail/actionDefinition.ts (hunk 3 above)
Comment (JusteenRandev) [reply packages/destination-actions/src/destinations/engage/sendgrid/sendEmail/actionDefinition.ts:126]: There isn't any explicit reason.

**Comment by joe-ayoub-segment** [comment]
hi @JusteenRandev this PR has been deployed.

**Pull request merged**